### PR TITLE
Setting GitHub organization token flow for Team Admins

### PR DESCRIPTION
**Pull request by gokmen**
screencast: http://recordit.co/bNsxwNqJ0e

This will provide the flow for GitHub integration and keeping the admin's token in `JGroupData` for future requirements.

completes: #9781 , #9904 

cc/ @devrim

### CODE REVIEW

**Comment by gokmen** [comment]
There is an ongoing issue with my api changes, working on that.

File: client/app/lib/providers/computecontroller.ui.coffee [hunk 1]
```diff
@@ -381,6 +381,31 @@ module.exports = class ComputeControllerUI
             Are you sure you want to proceed?</p>
           '
           button  : 'Yes, remove'
+        enableTeamOAuth :
+          title   : 'Do you want to setup organization token?'
```
Comment (sinan) [inline client/app/lib/providers/computecontroller.ui.coffee]: 'Do you want to setup an organization token?'

File: client/app/lib/providers/computecontroller.ui.coffee [hunk 2]
```diff
@@ -381,6 +381,31 @@ module.exports = class ComputeControllerUI
...
+          message : applyMarkdown "
+            In order to use and share GitHub organization tokens with admins
+            you can set a team token for your team. You need to be organization
```
Comment (sinan) [inline client/app/lib/providers/computecontroller.ui.coffee]: > you can set a team token for your team. For this, you need to be organization

File: client/app/lib/providers/computecontroller.ui.coffee [hunk 3]
```diff
@@ -381,6 +381,31 @@ module.exports = class ComputeControllerUI
...
+            admin on GitHub and a team admin on Koding. \n\n
+
+            This will require you to allow `#{options.scope}` scopes with your
```
Comment (sinan) [inline client/app/lib/providers/computecontroller.ui.coffee]: > You will need to allow `#{options.scope}` scopes with your

File: client/app/lib/providers/computecontroller.ui.coffee [hunk 4]
```diff
@@ -381,6 +381,31 @@ module.exports = class ComputeControllerUI
...
+            integration. \n\n
+
+            Do you want to enable your GitHub integration with these scopes or
+            do you want to setup this integration as a regular user?
+          "
+          button  : 'Yes, Setup as Organization'
```
Comment (sinan) [inline client/app/lib/providers/computecontroller.ui.coffee]: > Setup as Organization

File: client/app/lib/providers/computecontroller.ui.coffee [hunk 5]
```diff
@@ -381,6 +381,31 @@ module.exports = class ComputeControllerUI
...
+          cancel  : 'No, Setup as User'
```
Comment (sinan) [inline client/app/lib/providers/computecontroller.ui.coffee]: Setup as User

File: client/app/lib/providers/computecontroller.ui.coffee [hunk 6]
```diff
@@ -381,6 +381,31 @@ module.exports = class ComputeControllerUI
...
+        disableTeamOAuth :
+          title   : 'Do you want to disable organization token?'
+          message : applyMarkdown "
+            Currently your GitHub token is also set as default for team
+            integration, if you continue this integration will be removed
```
Comment (sinan) [inline client/app/lib/providers/computecontroller.ui.coffee]: > integration, if you continue, this integration will be removed

File: client/home/lib/account/homeaccountintegrationsview.coffee [hunk 7]
```diff
@@ -46,22 +58,38 @@ module.exports = class HomeAccountIntegrationsView extends kd.CustomHTMLView
...
 
       @containers[provider].addSubView @switches[provider]
 
+      @containers[provider].addSubView @loaders[provider] = @getLoaderView()
+
+
+  handleForeignAuth: (provider) -> @fetchOAuthInfo =>
 
-  handleForeignAuth: (provider) ->
+    if @linked[provider]
```
Comment (sinan) [inline client/home/lib/account/homeaccountintegrationsview.coffee]: early `return` ?

**Review by sinan — APPROVED** [review]
only some copy changes necessary 👍

**Pull request merged**